src/components/structures/ContextMenu.tsx

```diff
@@ -238,52 +239,33 @@ export default class ContextMenu extends React.PureComponent<IProps, IState> {
     };
 
     private onKeyDown = (ev: React.KeyboardEvent) => {
-        // don't let keyboard handling escape the context menu
-        ev.stopPropagation();
-
-        if (!this.props.managed) {
-            if (ev.key === Key.ESCAPE) {
+        // If someone is managing their own focus, we will only exit for them with Escape.
+        // They are probably using props.focusLock along with this option as well.
+        if(!this.props.managed) {
+            if(ev.key === Key.ESCAPE) {
                 this.props.onFinished();
-                ev.preventDefault();
             }
             return;
         }
 
-        // only handle escape when in an input field
-        if (ev.key !== Key.ESCAPE && getInputableElement(ev.target as HTMLElement)) return;
-
-        let handled = true;
-
-        switch (ev.key) {
-            // XXX: this is imitating roving behaviour, it should really use the RovingTabIndex utils
-            // to inherit proper handling of unmount edge cases
-            case Key.TAB:
-            case Key.ESCAPE:
-            case Key.ARROW_LEFT: // close on left and right arrows too for when it is a context menu on a <Toolbar />
-            case Key.ARROW_RIGHT:
-                this.props.onFinished();
-                break;
-            case Key.ARROW_UP:
-                this.onMoveFocus(ev.target as Element, true);
-                break;
-            case Key.ARROW_DOWN:
-                this.onMoveFocus(ev.target as Element, false);
-                break;
-            case Key.HOME:
-                this.onMoveFocusHomeEnd(this.state.contextMenuElem, true);
-                break;
-            case Key.END:
-                this.onMoveFocusHomeEnd(this.state.contextMenuElem, false);
-                break;
-            default:
-                handled = false;
+        // When an <input> is focused, only handle the Escape key
+        if (getInputableElement(ev.target as HTMLElement) && ev.key !== Key.ESCAPE) {
+            return;
         }
 
-        if (handled) {
-            // consume all other keys in context menu
-            ev.preventDefault();
+        if (
+            ev.key === Key.ESCAPE ||
+            // You can only navigate the ContextMenu by arrow keys and Home/End (see RovingTabIndex).
+            // Tabbing to the next section of the page, will close the ContextMenu.
+            ev.key === Key.TAB ||
+            // When someone moves left or right along a <Toolbar /> (like the
+            // MessageActionBar), we should close any ContextMenu that is open.
+            ev.key === Key.ARROW_LEFT ||
+            ev.key === Key.ARROW_RIGHT
+        ) {
+            this.props.onFinished();
         }
-    };
+    }
```

Comment:
This function will now only determine whether to close the `ContextMenu` (via `this.props.onFinished();`).

All of the navigation is now delegated to `RoamingTabIndex`